### ISSUE TITLE
tf.io.decode_image will flip the image

### ISSUE THREAD

**Issue by undcloud** [tensorflow/tensorflow#51813]
Please look at [colab](https://colab.research.google.com/drive/1gybGlto8ol7lnE82G9LLx0-DhRWtweUd#scrollTo=gecIt24H9fHO)

![IMG_20210729_194824430_MFNR](https://user-images.githubusercontent.com/33624574/131965649-813d23db-d14d-4c12-8dd5-25c3bf313e90.jpg)

**Comment by sushreebarsa** [comment]
@AR-fan 
In order to expedite the trouble-shooting process,Can you please fill the issue [template](https://github.com/tensorflow/tensorflow/issues/new/choose).Thanks!

**Comment by AR-fan** [comment]
@sushreebarsa 

- Have I written custom code :  
```  
img_bytes = open("./IMG_20210729_194824430_MFNR.jpg", 'rb').read()
img = tf.io.decode_image(img_bytes)
new_img_bytes = tf.io.encode_jpeg(img)
```  
- OS Platform and Distribution : colab Linux version 5.4.104+   
- TensorFlow version : 2.6.0  
- Python version : Python 3.7.11  
- GCC/Compiler version (if compiling from source) : gcc (Ubuntu 7.5.0-3ubuntu1~18.04) 7.5.0  
- CUDA/cuDNN version : None  

**Describe the current behavior**  
tf.io.decode_image op will flip the input image  

**Describe the expected behavior**  
tf.io.decode_image op not flip the input image  

- Do you want to contribute a PR? (yes/no):yes

**Comment by sushreebarsa** [comment]
@AR-fan I tried to run your code on Colab using TF v2.6 and didn't face the issue reported here ,Please find the [gist](https://colab.research.google.com/gist/sushreebarsa/72f61f633b7a0b5bad7e5aacd60561f2/-github_issue_41126_ipynb.ipynb#scrollTo=CfrODBN8a4ul) for reference. Could you please have a look at the [link](https://www.tensorflow.org/api_docs/python/tf/io/decode_image) and similar [issue](https://github.com/tensorflow/tensorflow/issues/41126)? Thank you!

**Comment by AR-fan** [comment]
@sushreebarsa 
Please try this picture:
![IMG_20210729_194824430_MFNR](https://user-images.githubusercontent.com/33624574/132168738-77702ed9-66e4-481a-9489-605554e40931.jpg)

**Comment by AR-fan** [comment]
@sushreebarsa 
I have seen that issue, but not works for me.  
Using tf op is right in most case, but fail in this case.

**Comment by sushreebarsa** [comment]
@AR-fan Could you please refer to the [gist ](https://colab.research.google.com/gist/sushreebarsa/74683afc831c6bf99b4c728f1f4fe5f3/untitled433.ipynb) and let us know if it helps? Thank you!

**Comment by AR-fan** [comment]
@sushreebarsa 
Could you please have a look at this [gist](https://colab.research.google.com/drive/1gybGlto8ol7lnE82G9LLx0-DhRWtweUd?usp=sharing)?    
I modify your gist and use wget to get picture.  
The result is still wrong.    
If you find a solution , please tell me , thanks~
Onz

**Comment by sushreebarsa** [comment]
@sanatmpa1 Was able to reproduce the issue on colab using TF 2.6 and tf-nightly ,Please find the gist[ here ](https://colab.research.google.com/gist/sushreebarsa/7d4afd700b35821cf840464ec8eb67c9/-github_issue_41126_ipynb.ipynb#scrollTo=11TT3PoSVUuZ) for reference.Thanks!

**Comment by AR-fan** [comment]
hi, how is it going?~

**Comment by jvishnuvardhan** [comment]
@AR-fan When I used two functions from a TF tutorial https://www.tensorflow.org/tutorials/generative/style_transfer to (load & process) and show image, original image and decoded image show similarly. Can you please verify whether `cv` library changing the order of the image?

Please check the [gist here](https://colab.research.google.com/gist/sushreebarsa/7d4afd700b35821cf840464ec8eb67c9/-github_issue_41126_ipynb.ipynb)

**Comment by AR-fan** [comment]
@jvishnuvardhan 
I have collected a lot of pictures. I need pictures not to be reversed. I found that using tf.io will reverse some pictures, but some pictures will not be reversed, but opencv will never reverse.
Maybe I'll use OpenCV to process the picture, because it will be done as I expected.

**Comment by sushreebarsa** [comment]
@undcloud Could you please have a look at this [gist](https://colab.research.google.com/gist/sushreebarsa/59786358dbb033761ecb42d85fc9a5e1/-github_issue_41126_ipynb.ipynb#scrollTo=IxFFU4TCZzvY), the issue seems to be fixed in the latest TF v2.13. 
Thank you!
